### ISSUE TITLE
ray_implicit_intersection returns a spurious near-origin crossing when the ray starts far from the SDF band

### ISSUE THREAD

**Issue by swahtz** [openvdb/fvdb-core#692]
## Summary

`GridBatch.ray_implicit_intersection` / `Grid.ray_implicit_intersection` returns an
**incorrect** intersection for a ray that starts far from the SDF's active narrow band. The
reported hit collapses to a spurious point ~¼ of a voxel from the ray origin instead of the
true zero-crossing. Moving the ray origin close to the surface returns the correct result.

This silently breaks ray-marching an SDF from a distant camera/emitter — a common setup
(e.g. `mitre_example/depth_to_sdf.py`, level-set raytracing/rendering).

## Reproduction (self-contained, no data files)

Builds an eikonal SDF the same way a TSDF-fusion pipeline does (`integrate_tsdf` +
`reinitialize_sdf`) on a synthetic sloped depth image, then traces the central pixel's ray
straight down `+z` from several start heights:

```python
import math
import torch
import fvdb

device = "cuda" if torch.cuda.is_available() else "cpu"
H = W = 96
vs = 460.0            # voxel size (metres)
trunc = vs * 4.0      # TSDF truncation distance

# Synthetic depth image: surface depth ramps across the frame (terrain-like relief),
# viewed by a pinhole camera at the world origin looking down +z.
u = torch.arange(W, device=device).float()
depth = (460.0 + (u / (W - 1)) * (5500.0 - 460.0)).unsqueeze(0).repeat(H, 1)  # (H, W)
fx = fy = W / (2.0 * math.tan(math.radians(30.0)))
cx = cy = W / 2.0

proj = torch.tensor([[fx, 0, cx], [0, fy, cy], [0, 0, 1]], device=device).unsqueeze(0)
cam_to_world = torch.eye(4, device=device).unsqueeze(0)
grid = fvdb.GridBatch.from_zero_voxels(device=device, voxel_sizes=vs, origins=0.0)
tsdf0 = grid.jagged_like(torch.zeros(grid.total_voxels, device=device))
w0 = grid.jagged_like(torch.zeros(grid.total_voxels, device=device))
grid, tsdf, _ = grid.integrate_tsdf(trunc, proj, cam_to_world, tsdf0, w0, depth.unsqueeze(0))
sdf = grid.jagged_like(tsdf.jdata * trunc)
sdf = grid.reinitialize_sdf(sdf, band=max(1, round(trunc / vs)))

true_z = depth[H // 2, W // 2].item()   # central ray hits the surface at world z == its depth
d = torch.tensor([[0.0, 0.0, 1.0]], device=device)
print(f"grid: {grid.total_voxels} voxels; central-ray true surface world z = {true_z:.0f}")
for oz in [0.0, 1500.0, 2500.0]:
    o = torch.tensor([[0.0, 0.0, oz]], device=device)
    t = grid.ray_implicit_intersection(
        fvdb.JaggedTensor(o), fvdb.JaggedTensor(d), fvdb.JaggedTensor(sdf.jdata)
    ).jdata.item()
    tag = "OK" if abs((oz + t) - true_z) < 2 * vs else "WRONG"
    print(f"  ray origin z={oz:7.1f}:  t={t:9.2f}  ->  hit world z={oz + t:9.2f}   [{tag}]")
```

### Output

```
grid: 4887 voxels; central-ray true surface world z = 3007
  ray origin z=    0.0:  t=   115.00  ->  hit world z=   115.00   [WRONG]
  ray origin z= 1500.0:  t=  1442.82  ->  hit world z=  2942.82   [OK]
  ray origin z= 2500.0:  t=   442.82  ->  hit world z=  2942.82   [OK]
```

Same ray, same grid, same SDF — only the start position changes. The true surface is at
`z ≈ 3007`. From the origin the op returns `z ≈ 115`; from nearer the surface it resolves
correctly (`≈ 2942`). The bogus distance is a near-constant `~115 m ≈ ¼ voxel` here
regardless of the true surface depth — i.e. the crossing is detected essentially at the ray
origin.

## Expected

A ray that starts outside the surface should report the **first zero-crossing of the SDF
along the ray**, independent of how far the origin is from the active band — matching the
semantics of `nanovdb::math::zeroCrossing` (which the op's own comment says it mirrors).

## What I've narrowed down

- Op: `src/fvdb/detail/ops/RayImplicitIntersection.cu` (`rayImplicitCallback`). It transforms
  the ray to index space, clips to the grid's active bbox (`rayVox.clip(dualBbox)`, ~L81),
  walks **only active** voxels via `HDDALeafVoxelIterator`, and seeds the sign reference from
  "the first valid (non-NaN) voxel along the ray." (No out-of-bounds read — the iterator
  yields active voxels only, so `getValue(ijk) - 1` is in bounds.)
- The bbox clip does not help: the camera at the origin is *inside* the grid's bbox, so the
  long empty stretch between it and the band is interior, not clipped away.
- **The failure is data-dependent.** A cleanly, directly-constructed narrow-band SDF —
  flat *or* sloped — does **not** reproduce it, even from a distant origin. It reproduces only
  on SDFs produced by `integrate_tsdf` + `reinitialize_sdf` over a spatially-varying surface
  (as above). So the trigger involves the eikonal band's actual value/topology structure
  combined with a long empty prefix, not merely geometry.
- For contrast, nanovdb's `ex_raytrace_level_set` uses `nanovdb::math::zeroCrossing` on a
  classic level set where inactive voxels carry the signed background value, so the sign field
  is defined everywhere along the ray and the first crossing is found robustly regardless of
  eye distance.

## Impact

Silent incorrect results (no crash/error). Any "ray-march the SDF" use with the camera far
from the surface band is affected.

## Suggested direction

Make the long-empty-prefix traversal not latch an early crossing — e.g. seed the sign
reference only once the ray has reached the active band (advance to / clip against the first
active voxel), and double-check the `t0` / bracket bookkeeping in `rayImplicitCallback`
against a start that begins inside the band (which works).

## Environment

- fvdb `0.6.0`, nanovdb `32.9.1`, torch `2.11.0` / CUDA `13.0`
- GPU: NVIDIA RTX PRO 6000 Blackwell
- The op is unmodified by recent viewer work; reproduces independently of it.

**Comment by swahtz** [comment]
This is very likely something I introduced in my recent work on the ray intersection code.
